File: HansolYang/2022-07-29-Rotate-Image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        if (matrix.size() <= 1) {
            return;
        }
        
        int n = matrix.size() - 1;
        
        vector<vector<int>> temp = matrix;
        
        for (int i = 0; i <= n; i++) {
            for (int j = 0; j <= n; j++) {
                matrix[j][n-i] = temp[i][j];
            }
        }
        
    }
};

File: Jonghyeok Kim/2022-07-29-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        l, r = 0, len(matrix)-1
        while l < r:
            for i in range(r-l):
                top, bottom = l, r
                
                topLeft = matrix[top][l+i]
                matrix[top][l+i] = matrix[bottom-i][l]
                matrix[bottom-i][l] = matrix[bottom][r-i]
                matrix[bottom][r-i] = matrix[top+i][r]
                matrix[top+i][r] = topLeft
            r -= 1
            l += 1
        